webui/src/pages/repositories/repository/objects.jsx

```diff
@@ -200,7 +200,20 @@ const UploadButton = ({config, repo, reference, path, onDone, onClick, onHide, s
             inProgress: true
         })
         try {
-            await objects.upload(repo.id, reference.id, textRef.current.value, fileRef.current.files[0])
+            if (config.pre_sign_support && config.pre_sign_support_ui) {
```

Comment:
The UI should check just `pre_sign_support_ui` and the server should enable `pre_sign_support_ui` just if `pre_sign_support` is enabled.

Reply:
done